content/reference/interceptors.adoc

```diff
@@ -133,9 +133,21 @@ expect an asynchronous response.
             (if (:session context)
               context
               (go
-                (assoc context :auth-response (call-auth-system context))))})
+                (assoc context :auth-response (<! (call-auth-system context)))))})
 ----
 
+In this hypothetical example, the `go` block will invoke the
+authentication system and park, waiting for a response from the auth system.
+
+Pedestal will see that the interceptor returned a channel, and not a context map,
+and will free up the web server thread to handle other incoming requests
+while the `go` block is operating.
+
+Eventually, the auth system will convey a response, and the `go` block will
+complete, conveying the updated context through its channel.
+Pedestal will resume processing of the request, passing this updated
+context to the next interceptor in the interceptor chain.
+
```

Comment:
I'm glad you like the bit of narrative but it doesn't work if `call-auth-system` immediately returns a value. So I really think we need the non-blocking take here for the text to make sense.  And if we weren't dong something truly async, there wouldn't need to be a go block at all.